src/web/reducers/flows.js

```diff
@@ -361,6 +377,87 @@ reducer = reduceReducers(
         flowsByName: _.omit(state.flowsByName, name)
       }),
 
+      // Inserting a new skill essentially:
+      // 1. creates a new flow
+      // 2. creates a new "skill" node
+      // 3. puts that new node in the "insert buffer", waiting for user to place it on the canvas
+      [insertNewSkill]: (state, { payload }) => {
+        const skillId = payload.skillId.replace(/^botpress-skill-/i, '')
+        const flowRandomId = nanoid(5)
+        const flowName = `skills/${skillId}-${flowRandomId}.flow.json`
+
+        const newFlow = Object.assign(payload.generatedFlow, {
+          skillData: payload.data,
+          name: flowName,
+          location: flowName
+        })
+
+        const newNode = {
+          id: 'skill-' + flowRandomId,
+          type: 'skill-call',
+          skill: skillId,
+          name: `${skillId}-${flowRandomId}`,
+          flow: flowName,
+          next: payload.transitions || [],
+          onEnter: null,
+          onReceive: null
+        }
+
+        return {
+          ...state,
+          currentDiagramAction: 'insert_skill',
+          nodeInBuffer: newNode,
+          flowsByName: {
+            ...state.flowsByName,
+            [newFlow.name]: newFlow
+          }
+        }
+      },
+
+      [updateSkill]: (state, { payload }) => {
+        const modifiedFlow = Object.assign(state.flowsByName[payload.editFlowName], payload.generatedFlow, {
```

Comment:
This mutates `state.flowsByName[payload.editFlowName]`. If you want to not do that (really we should never ever mutate state, even if we'l replace the same key later) we can either start with an empty object (`Object.assign({}, x, y)`) or use spread which does exactly the same:
```
const modifiedFlow = {
  ...state.flowsByName[payload.editFlowName],
  ...payload.generatedFlow,
  flowData: payload.data,
  name: payload.editFlowName,
  location: payload.editFlowName
}
```

Similar above we should better not modify props of the function arguments (payload)